test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -4208,46 +4238,77 @@ def _load_configuration(self, infile, objtype=None):
             for k, v in conf.items():
-                fn = self.du.create_temp_file()
-                with open(fn, 'w') as fd:
-                    # handle server data saved as output of qmgr commands
-                    # by piping data back into qmgr
-                    if k.startswith('qmgr_'):
-                        qmgr = os.path.join(self.client_conf['PBS_EXEC'],
-                                            'bin', 'qmgr')
-                        fd.write("\n".join(v))
-                        self.du.run_cmd(self.hostname, [qmgr], cstdin=fd,
-                                        sudo=True)
-                    else:
-                        fd.write("\n".join(v))
-                        # append the last line
-                        fd.write("\n")
-                        self.du.run_cmd(self.hostname, ['cp', fn, k],
-                                        sudo=True)
-                    os.remove(fn)
-
+                # Load server configuration
+                if k.startswith('qmgr_'):
+                    fpath = self.du.create_temp_file()
+                    print_svr = '\n'.join(v)
+                    with open(fpath, 'w') as f:
+                        f.write(print_svr)
+                    file_qmgr = open(fpath)
+                    cmd = [qmgr]
+                    d = self.du.run_cmd(
+                        self.hostname, cmd, stdin=file_qmgr, sudo=True,
+                        logerr=False, level=logging.DEBUG)
+                    file_qmgr.close()
+                # Load pbs.conf file
+                elif k == "pbs_conf":
+                    cfg_vals = base64.decodestring(v)
+                    config = ast.literal_eval(cfg_vals)
+                    self.du.set_pbs_config(self.hostname, confs=config)
+                # Load hooks
+                elif k == "hooks":
+                    fpath = self.du.create_temp_file()
+                    print_hooks = '\n'.join(v)
+                    with open(fpath, 'w') as f:
+                        f.write(print_hooks)
+                    file_qmgr = open(fpath)
+                    cmd = [qmgr]
+                    d = self.du.run_cmd(
+                        self.hostname, cmd, stdin=file_qmgr, sudo=True,
+                        logerr=False, level=logging.DEBUG)
+                    file_qmgr.close()
+            nodes = conf['pbsnodes']
+            for node in nodes:
+                node_name = str(node['id'])
+                self.create_pbsnode(node_name, node)
             return True
         return False
 
+    def create_pbsnode(self, node_name, attrs):
+        """
+        Create node in PBS with given attributes
+        """
+        qmgr = os.path.join(self.client_conf['PBS_EXEC'],
+                            'bin', 'qmgr')
+        execcmd = "create node " + node_name
+        cmd = [qmgr, "-c", execcmd]
+        self.du.run_cmd(
+            self.hostname, cmd, sudo=True)
+        skip_atb_list = ['id', 'Port', 'pbs_version', 'pcpus',
```

Comment:
we can't ignore port and state, as in case of multi mom on a single host, the port will be different.

Reply:
Changed

Reply:
@hiren 
1. I created multiple moms on a single host and on different port using pbs_config,
pbs_conf --multi-mom --create=4 --init-port=60000
2. Deleted one of the moms & tried to create it back using qmgr command
3. I was getting an error in create node
qmgr -c "c n spartan-6 Port=60006"
qmgr obj=spartan-6 svr=default: Unknown node.

Whereas, when i run the same command for a node on different host it gets created successfully.
qmgr -c "c n 10.75.37.51 Port=60006".
Looks like it will fail in create node and will not go ahead with set node for multi mom on single host.

Reply:
@RKORavi You tagged wrong hiren :) my username is @hirenvadalia

Can you provide me pbs_config output with -lDEBUG when u created multiple moms.

Reply:
Sorry my bad, corrected it now :)

Reply:
@RKORavi can you please try this commands:
`pbs_config --multi-mom --create=4 --init-port=60000`
`qmgr -c "d n spartan-6"`
`qmgr -c "c n spartan-6 port=60006,Mom=spartan"`